### ISSUE TITLE
Focus trap not working with portals

### ISSUE THREAD

**Issue by nahumzs** [acl-services/paprika#351]
# Bug Report

## Prerequisites
**This is just a checklist, please delete this section after**

Since the Popover gets added at the end of the DOM, and the Takeover hijacks behaviour outside of its scope, a Popover within a Takeover doesn't work very well.

Notes ✏️
The Takeover uses focus-trap-react, which uses focus-trap.
The Popover gets added at the end of the DOM -- outside of the Takeover's HTML element.
When you open a popover, focus goes to its content: https://github.com/acl-services/paprika/blob/master/packages/Popover/src/Popover.js#L357

Except that the Takeover says I dont think so! and takes back the focus: https://github.com/davidtheclark/focus-trap/blob/master/index.js#L253
This causes the takeover to open quickly, then close, and focus moves to a weird spot.

Interacting with a Popover with the isEager or shouldKeepFocus prop:
The Popover opens, but the Takeover prevents you from interacting with its contents with the mouse or keyboard.

## Expected behavior
The user should be able to open any popover or component rendering a portal and focus trap shouldn't steal their focus. 

## Current behavior
Please visit this PR to get more information about it. https://github.com/acl-services/paprika/pull/314

## Screenshots / Gifs / Codepens
referred to the previous section
